src/main/java/com/ably/kafka/connect/ChannelConfig.java

```diff
@@ -0,0 +1,76 @@
+package com.ably.kafka.connect;
+
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ChannelMode;
+import io.ably.lib.types.ChannelOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CHANNEL_CONFIG;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_CIPHER_KEY;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_PARAMS;
+
+interface ChannelConfig {
+    String getName();
+
+    ChannelOptions getOptions() throws ChannelSinkConnectorConfig.ConfigException;
+}
+
+class ChannelConfigImpl implements ChannelConfig {
+    private final ChannelSinkConnectorConfig sinkConnectorConfig;
+
+    public ChannelConfigImpl(ChannelSinkConnectorConfig sinkConnectorConfig) {
+        this.sinkConnectorConfig = sinkConnectorConfig;
+    }
+
+    @Override
+    public String getName() {
+        return sinkConnectorConfig.getString(CHANNEL_CONFIG);
+    }
+
+    public ChannelOptions getOptions() throws ChannelSinkConnectorConfig.ConfigException {
+        final Logger logger = LoggerFactory.getLogger(ChannelSinkConnectorConfig.class);
+        ChannelOptions opts;
+        String cipherKey = sinkConnectorConfig.getString(CLIENT_CHANNEL_CIPHER_KEY);
+
+        if (cipherKey != null && !cipherKey.trim().isEmpty()) {
+            try {
+                opts = ChannelOptions.withCipherKey(cipherKey);
+            } catch (AblyException e) {
+                logger.error("Error configuring channel cipher key", e);
+                throw new ChannelSinkConnectorConfig.ConfigException("Error configuring channel cipher key", e);
+            }
+        } else {
+            opts = new ChannelOptions();
+        }
+
+        // Since we're only publishing, set the channel mode to publish only
+        opts.modes = new ChannelMode[]{ChannelMode.publish};
+        opts.params = getParams(sinkConnectorConfig.getList(CLIENT_CHANNEL_PARAMS));
+        return opts;
+    }
+
+    private Map<String, String> getParams(List<String> params) throws ChannelSinkConnectorConfig.ConfigException {
+        final Logger logger = LoggerFactory.getLogger(ChannelSinkConnectorConfig.class);
+
+        Map<String, String> parsedParams = new HashMap<String, String>();
```

Comment:
```suggestion
        final Map<String, String> parsedParams = new HashMap<String, String>();
```

Reply:
These were all present in previous methods, so I did not really change them. But I will apply your suggestions 👍

Reply:
https://github.com/ably/kafka-connect-ably/pull/50/commits/5ab746257b1e1695cf79084ecca2c10f3180053a

Reply:
Thanks. I do think it's worth tidying up things like this when moving code around, given you're going to being adding git diff noise anyway. 🤷